event/parsing/eventValidator.go

```diff
@@ -0,0 +1,146 @@
+package parsing
+
+import (
+	"errors"
+	"fmt"
+	"regexp"
+	"time"
+
+	"github.com/xmidt-org/glaukos/event/client"
+	"github.com/xmidt-org/glaukos/event/queue"
+)
+
+// EventValidation is an interface for checking if wrp messages and events pass certain validations
+type EventValidation interface {
+	IsEventValid(client.Event) (bool, error)
+	IsWRPValid(queue.WrpWithTime) (bool, error)
+	GetEventCompareTime(client.Event) (time.Time, error)
+	GetWRPCompareTime(queue.WrpWithTime) (time.Time, error)
+	ValidateType(string) bool
+}
+
+// EventRule is the config struct for event validation
+type EventRule struct {
+	Regex          string
+	CalculateUsing string
+	ValidFrom      time.Duration
+}
+
+// eventValidator implements a metricparser.EventValidator interface.
+// Keeps a set of information/rules to see if an event or wrp fits these rules.
+type eventValidator struct {
+	regex          *regexp.Regexp
+	calculateUsing TimeLocation
+	timeValidation TimeValidation
+}
+
+var (
+	ErrInvalidEventType = errors.New("event type doesn't match")
+	ErrInvalidRegex     = errors.New("invalid regex")
+	ErrInvalidBootTime  = errors.New("invalid boot-time")
+	ErrInvalidBirthdate = errors.New("invalid birthdate")
+	ErrTimeParse        = errors.New("parsing error")
+)
+
+// NewEventValidation creates a new EventValidator from an EventRule
+func NewEventValidation(rule EventRule, validTo time.Duration, currentTime func() time.Time) (EventValidation, error) {
```

Comment:
I think one of the go tools suggested that returning an unexported struct might be annoying to use which was why I put the interface in this file and had the constructor return the interface instead of *eventValidator. But since the other package should only use eventValidator as the interface type, moving it there sounds good!